azure-auth-helper/src/main/java/com/microsoft/azure/auth/AzureAuthHelper.java

```diff
@@ -58,27 +149,19 @@ public static AzureCredential refreshToken(AzureEnvironment env, String refreshT
         if (StringUtils.isBlank(refreshToken)) {
             throw new IllegalArgumentException("Parameter 'refreshToken' cannot be empty.");
         }
-        final ExecutorService executorService = Executors.newSingleThreadExecutor();
-        try {
-            // TODO: handle proxy
-            final AuthenticationContext authenticationContext =
-                    new AuthenticationContext(baseURL(env), true, executorService);
-            return AzureCredential.fromAuthenticationResult(authenticationContext.acquireTokenByRefreshToken(refreshToken,
-                            Constants.CLIENT_ID, env.managementEndpoint(), null).get());
-        } catch (MalformedURLException | InterruptedException | ExecutionException e) {
-            throw new AzureLoginFailureException(e.getMessage());
-        } finally {
-            executorService.shutdown();
-        }
+
+        return new AzureCredentialCallable(baseURL(env), authenticationContext -> authenticationContext
```

Comment:
The indentation looks weird. The `acquireTokenByRefreshToken` and `call` should be at the same level.

Reply:
done